packages/ui/fe-bundle/src/utils/theme/loadDocsTypography.ts

```diff
@@ -0,0 +1,122 @@
+import type * as FernRegistryDocsRead from "@fern-fern/registry-browser/api/resources/docs/resources/v1/resources/read";
+
+export function getFontExtension(url: URL): string {
+    const ext = url.pathname.split(".").pop();
+    if (ext == null) {
+        throw new Error("No extension found");
+    }
+    return ext;
+}
+
+export type FontType = "headingsFont" | "bodyFont" | "codeFont";
+
+export interface GenerationFontConfig {
+    fontType: FontType;
+    fontName: string;
+    fontExtension: string;
+    fontUrl: FernRegistryDocsRead.Url | undefined;
+}
+
+export function getFontConfig(
+    docsDefinition: FernRegistryDocsRead.DocsDefinition,
+    fontType: FontType
+): GenerationFontConfig | undefined {
+    const typographyFontType = docsDefinition.config.typography?.[fontType];
+    if (typographyFontType == null) {
+        return undefined;
+    }
+    const fontName = typographyFontType.name;
+    const fontUrl = docsDefinition.files[typographyFontType.fontFile];
+    const fontExtension = getFontExtension(new URL(fontUrl ?? ""));
+
+    const fontConfig: GenerationFontConfig = {
+        fontType,
+        fontName,
+        fontExtension,
+        fontUrl,
+    };
+
+    return fontConfig;
+}
+
+/**
+ * Load Documentation Typography configuration
+ * @param docsDefinition
+ * @returns
+ */
+export function loadDocTypography(docsDefinition: FernRegistryDocsRead.DocsDefinition): GenerationFontConfig[] {
```

Comment:
This API of returning `GenerationFontConfig[]` isn't ideal because it can represent invalid states (e.g. multiple headingsFont values). nbd but imo better to have an interface like this:
```ts
interface GenerationFontConfigs {
  headingsFont: GenerationFontConfig | undefined;
  bodyFont: GenerationFontConfig | undefined;
  codeFont: GenerationFontConfig | undefined;
}
```

with that approach it's impossible to represent invalid states